src/directives.adoc

```diff
@@ -0,0 +1,308 @@
+* The `$ion_symbol_table` struct and its behavior are subsumed by a new `$ion_encoding` top-level
+S-expression that imports and defines modules, then separately assembles the local symbol and
+macro tables.
+
+This chapter focuses on the new components of the top-level context, the `$ion_encoding`
+S-expression that controls it, and the new parts of shared symbol tables that effect shared
+modules.
+// TODO The inner workings of modules are covered by TODO and macros by TODO
+
+=== Document Structure
+
+TODO Cover document segmentation, environment components, etc.
+
+TODO The below should probably move elsewhere
+
+As we’ve seen, encoding directives manipulate the global context, managing modules and installing
+(some of) them into the local symbol and macro tables.
+To clarify this behavior, we should first discuss the lifecycle of modules.
+An Ion 1.1 implementation must manage a few distinct sets of modules:
+
+* The _loaded modules_ are those that have been defined or loaded by an encoding directive, or
+transitively loaded from another loaded module.
+* The _available modules_ are loaded modules that have been assigned a name via a `load` or
+`import` clause.
+* The _installed modules_ are available modules that have been listed in an encoding directive's
+`symbol_table` or `macro_table` field.
+Technically, modules are installed immediately following termination of the `$ion_encoding` directive.
+
+Each encoding directive on the stream fully replaces the prior context.
+A user module becomes unavailable when a succeeding directive fails to retain it explicitly.
+A loaded module can be unloaded (garbage collected) when its no longer reachable from an
+available module.
+
+=== Ion Version Markers
+
+The bootstrap directive, required at the start of all Ion 1.1 segments, and acceptable
+mid-stream, is the Ion version marker:
+
+[{nrm}]
+----
+*$ion_1_1*
+----
+
+This keyword has the effect of resetting the encoding context to the default modules, symbols, and
+macros provided by the Ion specification.
+More precisely, the default context has a single available module named `$ion`, installed for
+both symbols and macros.
+This ensures that the system symbols and system macros provided by Ion 1.1 are available by
+default.
+
+The system module and its macros are in fact available everywhere in the document, and cannot be
+removed or redefined by `$ion_encoding`: to a large degree, it's as if the `*retain*`,
+`*symbol_table*`, and `*macro_table*` clauses all have `$ion` as their implicit first element.
+As a result, system macros can always be invoked by `(:$ion:__name__ …)`.
+
+System macros have one additional bit of special handling: they are binary-encoded using a
+dedicated opcode, using a dedicated address space that’s independent of the explicitly-enumerated
+modules in `*macro_table*`.
+This means that the initial range of unqualified numeric macro references like `(:3 …)` don’t
+inherently refer to system macros.
```

Comment:
Yeah you got the nuance.  I def need to find a better way to explain how system macros are handled.  It's tricky because macro address allocation works differently than symbols in a couple ways now.   Cut #264